plugins/out_azure_logs_ingestion/azure_logs_ingestion.c

```diff
@@ -0,0 +1,445 @@
+                                &final_payload, &final_payload_size);
+        if (ret == -1) {
+            flb_plg_error(ctx->ins,
+                          "cannot gzip payload, disabling compression");
+        }
+        else {
+            is_compressed = FLB_TRUE;
+            flb_plg_debug(ctx->ins, "enabled payload gzip compression");
+            /* JSON buffer will be cleared at cleanup: */
+        }
+    }
+
+    /* Compose HTTP Client request */
+    c = flb_http_client(u_conn, FLB_HTTP_POST, ctx->dce_u_url,
+                        final_payload, final_payload_size, NULL, 0, NULL, 0);
+
+    if (!c) {
+        flb_plg_warn(ctx->ins, "retrying payload bytes=%d", flb_sds_len(final_payload_size));
+        flush_status = FLB_RETRY;
+        goto cleanup;
+    }
+
+    /* Append headers */
+    flb_http_add_header(c, "User-Agent", 10, "Fluent-Bit", 10);
+    flb_http_add_header(c, "Content-Type", 12, "application/json", 16);
+    if (is_compressed) {
+        flb_http_add_header(c, "Content-Encoding", 16, "gzip", 4);
+    }
+    flb_http_add_header(c, "Authorization", 13, token, flb_sds_len(token));
+    flb_http_buffer_size(c, FLB_HTTP_DATA_SIZE_MAX);
+
+    /* Execute rest call */
+    ret = flb_http_do(c, &b_sent);
+    if (ret != 0) {
+        flb_plg_warn(ctx->ins, "http_do=%i", ret);
+        flush_status = FLB_RETRY;
+        goto cleanup;
+    }
+    else {
+        if (c->resp.status >= 200 && c->resp.status <= 299) {
+            flb_plg_info(ctx->ins, "http_status=%i, dcr_id=%s, table=%s",
+                         c->resp.status, ctx->dcr_id, ctx->table_name);
+            flush_status = FLB_OK;
+            goto cleanup;
+        }
+        else {
+            if (c->resp.payload_size > 0) {
+                flb_plg_warn(ctx->ins, "http_status=%i:\n%s",
+                             c->resp.status, c->resp.payload);
+            }
+            else {
+                flb_plg_warn(ctx->ins, "http_status=%i", c->resp.status);
+            }
+            flb_plg_debug(ctx->ins, "retrying payload bytes=%d", flb_sds_len(final_payload_size));
+            flush_status = FLB_RETRY;
+            goto cleanup;
+        }
+    }
+
+cleanup:
+    /* cleanup */
+    if (json_payload) {
+        flb_sds_destroy(json_payload);
+    }
+    /* release compressed payload */
+    if (is_compressed == FLB_TRUE) {
+        flb_free(final_payload);
+    }
+
+    if (c) {
+        flb_http_client_destroy(c);
+    }
+    if (u_conn) {
+        flb_upstream_conn_release(u_conn);
+    }
+    FLB_OUTPUT_RETURN(flush_status);
+}
+
+static int cb_azure_logs_ingestion_exit(void *data, struct flb_config *config)
+{
+    struct flb_az_li *ctx = data;
+    flb_plg_debug(ctx->ins, "exiting logs ingestion plugin");
+    flb_az_li_ctx_destroy(ctx);
+    return 0;
+}
+
+/* Configuration properties map */
+static struct flb_config_map config_map[] = {
+    {
+     FLB_CONFIG_MAP_STR, "tenant_id", (char *)NULL, 0, FLB_TRUE,
```

Comment:
for config maps use what other plugins does to set up the variables in the config map, I am referring to the parameters per line so it gets easier to read, this is an example from Loki:

```c
    {
     FLB_CONFIG_MAP_CLIST, "remove_keys", NULL,
     0, FLB_TRUE, offsetof(struct flb_loki, remove_keys),
     "Comma separated list of keys to remove."
    },
```

Reply:
change the options per line for all config map entries

Reply:
Fixed all config lines to follow the same pattern